### ISSUE TITLE
ChakraCore shell fails to run dist/run.js

### ISSUE THREAD

**Issue by bmeurer** [bmeurer/web-tooling-benchmark#21]
Somehow ChakraCore is failing to execute the generated dist file, i.e.

```bash
$ ~/Applications/ChakraCore-v1.7.3/bin/ch dist/run.js
Running Web Tooling Benchmark 0.1.0...
--------------------------------------
         acorn:  4.88 runs/sec  ±8.64%
         babel:  5.09 runs/sec  ±4.06%
       babylon:  7.29 runs/sec  ±3.68%
         buble:  6.29 runs/sec  ±2.37%
          chai:  1.95 runs/sec  ±3.12%
Encountered error, aborting benchmark...
  coffeescript:  0.00 runs/sec  ±0.00%
--------------------------------------
Geometric mean:  5.11 runs/sec
```

We should also improve the error reporting here. And then find out why only `ch` is failing on this, while all other shells pass.

**Comment by bmeurer** [comment]
The error that I get is:

```
Encountered error running benchmark coffeescript, aborting...
TypeError: Unable to get property 'name' of undefined or null reference
   at Param.prototype.eachName (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:36965:26)
   at Code.prototype.eachParamName (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:36831:11)
   at Code.prototype.compileNode (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:36558:9)
   at Base.prototype.compileToFragments (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:32981:9)
   at Block.prototype.compileNode (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:33555:13)
   at Base.prototype.compileToFragments (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:32981:9)
   at Block.prototype.compileToFragments (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:33525:11)
   at Class.prototype.compileClassDeclaration (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:35367:11)
   at Base.prototype.compileToFragments (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:32981:9)
   at Assign.prototype.compileNode (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:36074:9)
   at Base.prototype.compileToFragments (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:32981:9)
   at Class.prototype.compileNode (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:35335:11)
   at Base.prototype.compileToFragments (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:32981:9)
   at Assign.prototype.compileNode (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:36074:9)
   at Base.prototype.compileToFragments (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:32981:9)
   at Block.prototype.compileNode (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:33555:13)
   at Block.prototype.compileWithDeclarations (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:33635:9)
   at Block.prototype.compileRoot (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:33603:9)
   at Block.prototype.compileToFragments (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:33527:11)
   at Anonymous function (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:30058:5)
   at Anonymous function (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:29984:9)
   at module.exports.fn (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:29775:5)
   at Function code (Function code:3:112)
   at clock (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:10225:13)
   at clock (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:10399:7)
   at cycle (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:10588:9)
   at run (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:10695:13)
   at execute (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:9441:9)
   at invoke (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:9551:20)
   at compute (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:10547:7)
   at run (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:10700:11)
   at execute (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:9441:9)
   at invoke (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:9551:20)
   at runSuite (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:9750:7)
   at Anonymous function (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:8246:1)
   at __webpack_require__ (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:25:12)
   at Anonymous function (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:68:11)
   at Global code (/Users/bmeurer/Projects/WebToolingBenchmark/web-tooling-benchmark/dist/run.js:6:11)
```

**Comment by bmeurer** [comment]
@bterlson is going to take a look.

**Comment by bterlson** [comment]
@bmeurer it seems like a recent regression whatever it is, and will take a bit to get a root cause. In the meantime it seems to run in version 1.7.1 so you could use that version in the meantime.

**Comment by bterlson** [comment]
@bmeurer a fix for this issue was committed a few days ago (https://github.com/Microsoft/ChakraCore/commit/14df0346b2af4cd74e2e57d9c4c4785f11e19fd3), so you can also use master builds as well. This fix will be included in 1.7.4 as well.

**Comment by bmeurer** [comment]
Works like a charm now, thanks!

```bash
$ ~/Projects/3rdparty/ChakraCore/out/Release/ch dist/run.js
Running Web Tooling Benchmark 0.2.0...
--------------------------------------
         acorn:  3.99 runs/sec  ±4.74%
         babel:  4.41 runs/sec  ±2.84%
       babylon:  6.10 runs/sec  ±2.81%
         buble:  4.85 runs/sec  ±1.67%
          chai:  1.37 runs/sec  ±6.73%
  coffeescript:  0.17 runs/sec  ±3.65%
        espree:  3.67 runs/sec  ±4.22%
       esprima:  6.36 runs/sec  ±1.39%
        jshint:  3.42 runs/sec  ±1.15%
         lebab:  3.06 runs/sec  ±1.99%
       prepack:  4.39 runs/sec  ±2.46%
      prettier:  6.79 runs/sec  ±3.70%
    source-map:  9.44 runs/sec  ±3.04%
    typescript: 10.60 runs/sec  ±4.40%
     uglify-js:  4.02 runs/sec  ±3.04%
     uglify-es:  9.28 runs/sec  ±1.80%
--------------------------------------
Geometric mean:  3.94 runs/sec
```
